zebra/zebra_vxlan.c

```diff
@@ -9637,3 +9657,66 @@ static void zebra_vxlan_sg_cleanup(struct hash_backet *backet, void *arg)
 
 	zebra_vxlan_sg_del(vxlan_sg);
 }
+
+/************************** EVPN BGP config management ************************/
+static void zvni_evpn_cfg_cleanup(struct hash_bucket *bucket, void *ctxt)
+{
+        zebra_vni_t *zvni = NULL;
+	zebra_l3vni_t *zl3vni = NULL;
+
+        zvni = (zebra_vni_t *)bucket->data;
+	zvni->advertise_gw_macip = 0;
+	zvni->advertise_svi_macip = 0;
+	zvni->advertise_subnet = 0;
+
+	zl3vni = zl3vni_from_vrf(zvni->vrf_id);
+	if (zl3vni)
+	    listnode_delete(zl3vni->l2vnis, zvni);
+
+	zvni_neigh_del_all(zvni, 0, 0, DEL_ALL_NEIGH);
+	zvni_mac_del_all(zvni, 0, 0, DEL_ALL_MAC);
+	zvni_vtep_del_all(zvni, 0);
+	zvni_del(zvni);
+}
+
+/* Cleanup EVPN configuration of a specific VRF */
+static void zebra_evpn_vrf_cfg_cleanup(struct zebra_vrf *zvrf)
+{
+	zvrf->advertise_all_vni = 0;
+	zvrf->advertise_gw_macip = 0;
+	zvrf->advertise_svi_macip = 0;
+	zvrf->vxlan_flood_ctrl = VXLAN_FLOOD_HEAD_END_REPL;
+
+	zvrf->dup_addr_detect = false;
```

Comment:
dup_addr_detect is enabled by default (with default detection timers) in warning only mode. 
zvrf->dup_addr_detect = true;
zvrf->dad_max_move and zvrf->dad_freeze_time should be left with default values which will be passed when bgp instance is created.

Reply:
I have noticed BGP replaying the the DAD configuration when the BGP daemon came up. Did you mean to say that this happens in a specific mode only?. I built my FRR image with configure options mentioned in the FRRouting build instructions. pls let me know if i have missed something in here.

Reply:
When a bgp default instance comes up it will replay the configuration, that part is correct. The point is to retain default values of each field rather setting all to 0s would be ideal.